test/livebook/runtime/evaluator_test.exs

```diff
@@ -23,7 +23,8 @@ defmodule Livebook.Runtime.EvaluatorTest do
       send_to: self(),
       object_tracker: object_tracker,
       client_tracker: client_tracker,
-      ebin_path: ebin_path
+      ebin_path: ebin_path,
+      tmp_dir: "/tmp"
```

Comment:
We should make this tmpdir specific to the test. Also, it should be enabled explicitly with a tag, we can use the existing `@tag :with_ebin_path`. This way we can test the error when no tmpdir is available.

So you can change the setup block to this:

```elixir
setup ctx do
  {tmp_path, ebin_path} =
    if ctx[:with_ebin_path] do
      hash = ctx.test |> to_string() |> :erlang.md5() |> Base.encode32(padding: false)
      tmp_path = ["tmp", inspect(ctx.module), hash] |> Path.join() |> Path.expand()
      ebin_path = Path.join(tmp_path, "ebin")
      File.rm_rf!(ebin_path)
      File.mkdir_p!(ebin_path)
      Code.append_path(ebin_path)
      {tmp_path, ebin_path}
    end

  {:ok, object_tracker} = start_supervised(Evaluator.ObjectTracker)
  {:ok, client_tracker} = start_supervised(Evaluator.ClientTracker)

  opts = [
    send_to: self(),
    object_tracker: object_tracker,
    client_tracker: client_tracker,
    tmp_path: tmp_path,
    ebin_path: ebin_path
  ]

  {:ok, _pid, evaluator} = start_supervised({Evaluator, opts})

  %{
    evaluator: evaluator,
    object_tracker: object_tracker,
    client_tracker: client_tracker,
    tmp_path: tmp_path,
    ebin_path: ebin_path
  }
end
```

Then add `@tag :with_ebin_path` above each test that needs to have the tmpdir :)